src/status_im/data_store/realm/core.cljs

```diff
@@ -51,14 +57,35 @@
   (when realm
     (.close realm)))
 
-(defn migrate-realm [file-name schemas encryption-key]
-  (let [current-version (realm-version file-name encryption-key)]
-    (doseq [schema schemas
-            :when (> (:schemaVersion schema) current-version)
-            :let [migrated-realm (open-realm schema file-name encryption-key)]]
-      (close migrated-realm)))
+(defn reset-realm
+  "Delete realm & open a new database using encryption key"
+  [file-name schemas encryption-key]
+  (delete-realm file-name)
+  (open-realm (last schemas) file-name encryption-key))
+
+(defn- migrate-schemas
+  "Apply migrations in sequence and open database with the last schema"
+  [file-name schemas encryption-key current-version]
+  (doseq [schema schemas
+          :when (> (:schemaVersion schema) current-version)
+          :let [migrated-realm (open-realm schema file-name encryption-key)]]
+    (close migrated-realm))
   (open-realm (last schemas) file-name encryption-key))
 
+(defn migrate-realm
+  "Migrate realm if is a compatible version or reset the database"
+  [file-name schemas encryption-key]
+  (let [encrypted-version (encrypted-realm-version file-name encryption-key)
+        ;; If it's unencrypted reset schema
+        unencrypted?      (and (not encrypted-version)
+                               (unencrypted-realm? file-name))]
+    (cond
+      encrypted-version (migrate-schemas file-name schemas encryption-key encrypted-version)
```

Comment:
is it fine to call `migrate-schemas` if `encrypted-version` is -1?

Reply:
I believe so,  I haven't changed that logic + I have tested multiple times on clean installations

Reply:
I would add a comment though, because it _looks_ like this case with -1 was just forgotten.